models/lpjguess/R/met2model.LPJGUESS.R

```diff
@@ -0,0 +1,140 @@
+##' @param start_date the start date of the data to be downloaded (will only use the year part of the date)
+##' @param end_date the end date of the data to be downloaded (will only use the year part of the date)
+##' @param overwrite should existing files be overwritten
+##' @param verbose should the function be very verbose
+##' @author Istem Fer
+met2model.LPJGUESS <- function(in.path, in.prefix, outfolder, start_date, end_date, ..., overwrite=FALSE,verbose=FALSE){
+  
+  library(PEcAn.utils)
+  require(ncdf4)
+  require(lubridate)
+  
+  print("START met2model.LPJGUESS")
+  start_date <- as.POSIXlt(start_date, tz = "GMT")
+  end_date <- as.POSIXlt(end_date, tz = "GMT")
+  start_year <- year(start_date)
+  end_year <- year(end_date)
+  
+  year = sprintf("%04d",seq(start_year,end_year,1))
+  nyear = length(year) #number of years to simulate
+  
+  ## LPJ-GUESS looks for different input files for different climate variables
+  out.file <- out.files.full <- list()
+  var.names=c("tmp","pre","cld")
+  n.var=length(var.names)
+  long.names=c("air_temperature","precipitation_flux","surface_downwelling_shortwave_flux_in_air")
+  for(i in 1:n.var) out.file[[i]] <-paste(in.prefix, start_year, end_year, var.names[[i]], "nc", sep=".")
+  for(i in 1:n.var) out.files.full[[i]] <-file.path(outfolder,out.file[[i]])
+  
+  results <- data.frame(file = unlist(out.files.full),
+                        host = fqdn(),
+                        mimetype ='application/x-netcdf',
+                        formatname = 'lpj-guess.metfile' ,
+                        startdate = start_date ,
+                        enddate = end_date,
+                        dbfile.name = unlist(out.file),
+                        stringsAsFactors = FALSE)
+  print("internal results")
+  print(results)
+  
+  
+  ## check to see if the outfolder is defined, if not create directory for output
+  if(!file.exists(outfolder)){
+    dir.create(outfolder)
+  }
+
+    ## open netcdf files
+    ncin <- lapply(file.path(in.path,paste(in.prefix,year,"nc",sep=".")),nc_open)
+    
+    ## retrieve lat/lon 
+    lon=ncvar_get(ncin[[1]],"longitude")
+    lat=ncvar_get(ncin[[1]],"latitude")
+    
+    ## at least 2 lat-lon required for LPJ-GUESS to load the data
+    lon=c(lon,lon)
+    lat=c(lat,lat)
+    
+    ## calculate time step from the time-dimension length, check for leap year
+    tstep=ifelse(ncin[[1]]$dim$time$len%%365==0, ncin[[1]]$dim$time$len/365, ncin[[1]]$dim$time$len/366)
+    
+    ## read climate data
+    nc.tmp <- lapply(ncin, ncvar_get, long.names[1])
+    nc.pre <- lapply(ncin, ncvar_get, long.names[2])
+    nc.cld <- lapply(ncin, ncvar_get, long.names[3])
+    
+    ## aggregate to daily time steps, LPJ-GUESS reads daily climate data
+    tmp.list <- pre.list <- cld.list <- list()
+    for(y in 1:nyear){
+      if(leap_year(as.numeric(year[y]))) ind.vec=rep(1:366,each=tstep) else ind.vec=rep(1:365,each=tstep)
+      tmp.list[[y]] <- tapply(nc.tmp[[y]],ind.vec,mean)
```

Comment:
I think here is where you might be losing the 3 dimensions LPJ-GUESS needs to run.  I think you could either set up each level of tmp.list (etc) as an array with dims c(1, 1, time) or do in in your ncvar_put statement (I'll comment there too)